ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/electra/helpers/BeaconStateMutatorsElectra.java

```diff
@@ -184,19 +184,16 @@ public UInt64 computeConsolidationEpochAndUpdateChurn(
    * @param index validatorIndex
    */
   public void switchToCompoundingValidator(final MutableBeaconStateElectra state, final int index) {
-    if (PredicatesElectra.isEth1WithdrawalCredential(
-        state.getValidators().get(index).getWithdrawalCredentials())) {
-      final byte[] withdrawalCredentialsUpdated =
-          state.getValidators().get(index).getWithdrawalCredentials().toArray();
-      withdrawalCredentialsUpdated[0] = COMPOUNDING_WITHDRAWAL_BYTE;
-      state
-          .getValidators()
-          .update(
-              index,
-              validator ->
-                  validator.withWithdrawalCredentials(Bytes32.wrap(withdrawalCredentialsUpdated)));
-      queueExcessActiveBalance(state, index);
-    }
+    final byte[] withdrawalCredentialsUpdated =
+        state.getValidators().get(index).getWithdrawalCredentials().toArray();
+    withdrawalCredentialsUpdated[0] = COMPOUNDING_WITHDRAWAL_BYTE;
+    state
+        .getValidators()
+        .update(
+            index,
+            validator ->
+                validator.withWithdrawalCredentials(Bytes32.wrap(withdrawalCredentialsUpdated)));
+    queueExcessActiveBalance(state, index);
   }
```

Comment:
nit: probably not super important but this is super unsafe if the index doesn't exist, i wonder if we should put a comment in the description 
`PRE: validator must exist`
It's checked up in the process, but this is also public, and we probably want to be super careful where we attempt this...